packages/lesmis/src/api/queries/useEntityData.js

```diff
@@ -4,11 +4,30 @@
  *
  */
 import { useQuery } from 'react-query';
+import { useHistory } from 'react-router-dom';
 import { get } from '../api';
+import { useUser } from './useUser';
+
+const useValidatedQuery = query => {
+  const history = useHistory();
+  const { isLoggedIn } = useUser();
+
+  if (query.isError && query.error.code === 403) {
+    if (isLoggedIn) {
+      history.push('/not-found');
+    } else {
+      history.push('/login', { referer: history.location });
+    }
+  }
+  return query;
+};
 
 export const useEntityData = entityCode => {
-  return useQuery(['entity', entityCode], () => get(`entity/${entityCode}`), {
-    staleTime: 1000 * 60 * 60 * 1,
-    refetchOnWindowFocus: false,
-  });
+  return useValidatedQuery(
+    useQuery(['entity', entityCode], () => get(`entity/${entityCode}`), {
+      staleTime: 1000 * 60 * 60 * 1,
+      refetchOnWindowFocus: false,
+      retry: 2,
```

Comment:
What was the retry addition for?

Reply:
It's just a tweak based on a testing the not found behaviour. The default for the retry param in react-query is actually 3 which means that queries that fail are silently retried 3 times (with an exponential backoff delay). I think this is really nice default behaviour but I wanted it to redirect to the 404 page slightly quicker than that here.